avocado/utils/partition.py

```diff
@@ -0,0 +1,299 @@
+        except process.CmdError as error:
+            raise PartitionError(self, "Failed to mkfs", error)
+        else:
+            self.fstype = fstype
+
+    def mount(self, mountpoint=None, fstype=None, args=''):
+        """
+        Mount this partition to a mount point
+
+        :param mountpoint: If you have not provided a mountpoint to partition
+                object or want to use a different one, you may specify it here.
+        :param fstype: Filesystem type. If not provided partition object value
+                will be used.
+        :param args: Arguments to be passed to "mount" command.
+        """
+        if not mountpoint:
+            mountpoint = self.mountpoint
+        if not mountpoint:
+            raise PartitionError(self, "No mountpoint specified and no "
+                                 "default provided to this partition object")
+        if fstype is None:
+            fstype = self.fstype
+        else:
+            self.fstype = fstype
+
+        if self.mount_options:
+            args += ' -o ' + self.mount_options
+        if fstype:
+            args += ' -t ' + fstype
+        if self.loop:
+            args += ' -o loop'
+        args = args.lstrip()
+
+        with MtabLock():
+            if self.device in self.list_mount_devices():
+                raise PartitionError(self, "Attempted to mount mounted device")
+            if mountpoint in self.list_mount_points():
+                raise PartitionError(self, "Attempted to mount busy mountpoint")
+            if not os.path.isdir(mountpoint):
+                os.makedirs(mountpoint)
+            try:
+                process.system("mount %s %s %s"
+                               % (args, self.device, mountpoint), sudo=True)
+            except process.CmdError as details:
+                raise PartitionError(self, "Mount failed", details)
+        # Update the fstype as the mount command passed
+        self.fstype = fstype
+
+    def _unmount_force(self, mountpoint):
+        """
+        Kill all other jobs accessing this partition and force unmount it.
+
+        :return: None
+        :raise PartitionError: On critical failure
+        """
+        # Human readable list of processes
+        out = process.system_output("lsof " + mountpoint, ignore_status=True)
+        # Try to kill all pids
+        for pid in (line.split()[1] for line in out.splitlines()[1:]):
+            try:
+                process.system("kill -9 %s" % pid, ignore_status=True,
+                               sudo=True)
+            except OSError:
+                pass
+        # Unmount
+        try:
+            process.run("umount -f %s" % mountpoint, sudo=True)
+        except process.CmdError as details:
+            try:
+                process.run("umount -l %s" % mountpoint, sudo=True)
+            except process.CmdError as details:
+                raise PartitionError(self, "Force unmount failed", details)
+
+    def unmount(self, force=True):
```

Comment:
Basically the command is to unmount the partition. So the name unmount() is fine in my opinion.